book/en/src/by-example/channel.md

```diff
@@ -0,0 +1,112 @@
+# Communication over channels.
+
+Channels can be used to communicate data between running *software* tasks. The channel is essentially a wait queue, allowing tasks with multiple producers and a single receiver. A channel is constructed in the `init` task and backed by statically allocated memory. Send and receive endpoints are distributed to *software* tasks:
+
+```rust
+...
+const CAPACITY: usize = 5;
+#[init]
+    fn init(_: init::Context) -> (Shared, Local) {
+        let (s, r) = make_channel!(u32, CAPACITY);
+        receiver::spawn(r).unwrap();
+        sender1::spawn(s.clone()).unwrap();
+        sender2::spawn(s.clone()).unwrap();
+        ...
+```
+
+In this case the channel holds data of `u32` type with a capacity of 5  elements. 
+
+## Sending data
+
+The `send` method post a message on the channel as shown below:
+
+```rust
+#[task]
+async fn sender1(_c: sender1::Context, mut sender: Sender<'static, u32, CAPACITY>) {
+    hprintln!("Sender 1 sending: 1");
+    sender.send(1).await.unwrap();
+}
+```
+
+## Receiving data
+
+The receiver can `await` incoming messages:
+
+```rust
+#[task]
+async fn receiver(_c: receiver::Context, mut receiver: Receiver<'static, u32, CAPACITY>) {
+    while let Ok(val) = receiver.recv().await {
+        hprintln!("Receiver got: {}", val);
+        ...
+    }
+}
+```
+
+For a complete example:
+
+``` rust
+{{#include ../../../../rtic/examples/async-channel.rs}}
+```
+
+``` console
+$ cargo run --target thumbv7m-none-eabi --example async-channel --features test-critical-section 
+{{#include ../../../../rtic/ci/expected/async-channel.run}}
+```
+
+Also sender endpoint can be awaited. In case there the channel capacity has not been reached, `await` the sender can progress immediately, while in the case the capacity is reached, the sender is blocked until there is free space in the queue. In this way data is never lost.
```

Comment:
```suggestion
Also sender endpoint can be awaited. In case the channel capacity has not yet been reached, `await`-ing the sender can progress immediately, while in the case the capacity is reached, the sender is blocked until there is free space in the queue. In this way data is never lost.
```